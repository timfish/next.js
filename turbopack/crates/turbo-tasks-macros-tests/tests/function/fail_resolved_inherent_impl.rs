#![feature(arbitrary_self_types)]
#![allow(dead_code)]

use turbo_tasks::Vc;

#[turbo_tasks::value]
struct ExampleStruct;

#[turbo_tasks::value(transparent)]
struct IntegersVec(Vec<Vc<u32>>);

#[turbo_tasks::value_impl]
impl ExampleStruct {
    #[turbo_tasks::function(resolved)]
    fn return_contains_unresolved_vc(self: Vc<Self>) -> Vc<IntegersVec> {
        Vc::cell(Vec::new())
    }
}

fn main() {}
