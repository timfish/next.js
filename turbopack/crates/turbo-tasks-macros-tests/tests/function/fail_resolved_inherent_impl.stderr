error[E0277]: the trait bound `IntegersVec: ResolvedValue` is not satisfied
  --> tests/function/fail_resolved_inherent_impl.rs:14:29
   |
14 |     #[turbo_tasks::function(resolved)]
   |                             ^^^^^^^^ the trait `ResolvedValue` is not implemented for `IntegersVec`
   |
   = help: the following other types implement trait `ResolvedValue`:
             &T
             &mut T
             ()
             (A, Z, Y, X, W, V, U, T)
             (B, A, Z, Y, X, W, V, U, T)
             (C, B, A, Z, Y, X, W, V, U, T)
             (D, C, B, A, Z, Y, X, W, V, U, T)
             (E, D, C, B, A, Z, Y, X, W, V, U, T)
           and $N others
note: required by a bound in `assert_returns_resolved_value`
  --> tests/function/fail_resolved_inherent_impl.rs:14:29
   |
14 |     #[turbo_tasks::function(resolved)]
   |                             ^^^^^^^^ required by this bound in `assert_returns_resolved_value`
